website/dynamo.py

```diff
@@ -172,9 +172,13 @@ def get_item(self, table_name, key):
         return self._decode(result.get('Item', None))
 
     def query(self, table_name, key, reverse=False):
+        assert len(key.keys ()) == 1
+        key_field = list (key.keys ()) [0]
+        key_value = key [key_field]
         result = self.db.query(
             TableName=self.db_prefix + '-' + table_name,
-            Key = self._encode(key),
+            KeyConditionExpression = key_field + ' = :value',
+            ExpressionAttributeValues = {':value': {'S': key_value}},
```

Comment:
The `{'S': key_value }` will only work if the key is a string value. Now, that will probably be true 90% of the time, but it's still making me a little uneasy. We should have a proper DDB encoding function floating around in this module somewhere.

Can you use that instead?

Reply:
Definitely, great idea. Just replaced it with a call to `self.SERIALIZER.serialize`